src/rust-crypto/rust-crypto.ts

```diff
@@ -640,6 +642,43 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
         return new RustVerificationRequest(request, this.outgoingRequestProcessor, this.supportedVerificationMethods);
     }
 
+    /**
+     * Implementation of {@link CryptoApi#checkOwnCrossSigningTrust}
+     */
+    public async checkOwnCrossSigningTrust({ allowPrivateKeyRequests }: CheckOwnCrossSigningTrustOpts): Promise<void> {
+        // Download the device keys and add them into the olm machine
+        const queryResult = await this.downloadDeviceList(new Set([this.userId]));
+        await this.olmMachine.markRequestAsSent(
+            uuidv4(),
+            RustSdkCryptoJs.RequestType.KeysQuery,
+            JSON.stringify(queryResult),
+        );
```

Comment:
why is this necessary? making up a request id and telling the rust layer that we have sent that made-up request ID seems like an odd thing to do.

Reply:
well, first of all, doesn't the OlmMachine download the device keys automatically?

Reply:
Oh, you mean that I don't need the download the device list manually ? It's already done by the olm machine ?